### PR TITLE
#9 POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by almcw**
Request body accepts:

an object with the following properties:
username
body
Responds with:

the posted comment

### CODE REVIEW

File: naming-conventions.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Naming Conventions
```
Comment (jadelandeg) [inline naming-conventions.md:1]: at the end of the project you don't want this file in there

File: __tests__/app.test.js [hunk 2]
```diff
@@ -244,3 +244,52 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+
+  test("returns ERROR when passed article_id that doesn't exist", () => {
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send({ username: "icellusedkars", body: "great work!" })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No article found for article_id: 999");
+      });
+  });
```
[the remark this answers is not in the document]
Comment (almcw) [reply __tests__/app.test.js:274]: test added

File: naming-conventions.md (hunk 1 above)
Comment (almcw) [reply naming-conventions.md:1]: Yeah, I've taken it out now

**Comment by almcw** [comment]
Thanks @jadelandeg - error handling tweak and addition following your review

File: controllers/news.js [hunk 3]
```diff
@@ -49,8 +50,20 @@ exports.patchArticleVotes = (req, res, next) => {
...
 exports.getUsers = (req, res, next) => {
   selectUsers().then((users) => {
     res.status(200).send({ users });
   });
 };
+
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+  selectArticleById(article_id)
```
Comment (jadelandeg) [inline controllers/news.js:63]: you don't need this here, you're not doing anything with the result of it. Just do the post and it will be fine! :)

**Review by jadelandeg — APPROVED** [review]
tests look great, well done! Approving this, bonus test if you want to, but not necessary. Select new comment in the test and check that its been inserted :)

**Pull request merged**